### PR TITLE
[#207]Fixed Employee name wise sorting issue 

### PR DESCRIPTION
**Pull request by vaibhav8186**
Date of joining not showing correctly in Excel sheet.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -8,16 +8,10 @@ class UsersController < ApplicationController
...
-    if params[:status] == "all"
-      @usersxls = User.sort_by_id
-      @users = User.employees.skip(offset).limit(limit)
-    else
-      @usersxls = User.approved.sort_by_id
-      @users = User.approved.skip(offset).limit(limit)
-    end
+    offset = params[:offset] || 0
+    limit = params[:limit] || 50
+    @users = current_user.role == 'HR' || current_user.role == 'Admin' ? User.employees.offset(offset).limit(limit) : User.employees.approved.offset(offset).limit(limit)
```
Comment (pramodshinde) [inline app/controllers/users_controller.rb]: Can you remove these condition form here and add methods on user model like `user.hr?` etc if those methods are already not present?

Comment (vaibhav8186) [reply app/controllers/users_controller.rb]: yes. Used predefined method

File: app/assets/javascripts/users.js.coffee [hunk 2]
```diff
@@ -21,18 +21,36 @@ $(document).ready ->
...
+    d.status == "approved"
+  )
+  if($('#user_stream_table').length)
+    $("#user_stream_table").stream_table(options, data) if typeof data isnt "undefined"
 
+  $('#show').click ->
+    if $(this).text() == 'Show Approved'
+      $(this).text('Show All')
+      $('tr[hide]').removeAttr('hidden')
+      $('#show_text').text('Showing approved users only.')
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply app/assets/javascripts/users.js.coffee]: yes changed show_text to user_note_text

File: app/assets/javascripts/users.js.coffee [hunk 3]
```diff
@@ -21,18 +21,36 @@ $(document).ready ->
...
-    fields: (record) -> 
-      [
-        record.email,
-        record.public_profile.first_name,
-        record.public_profile.last_name
-      ].join('')
+    view: view,                  
+    data_url: '/users.json',
+    stream_after: 2,           
+    fetch_data_limit: 50
```
[the remark this answers is not in the document]
Comment (vaibhav8186) [reply app/assets/javascripts/users.js.coffee]: yes, we can increase this limit, now I have changed the limit to 200. So at a time, 200 records will fetch.

**Pull request merged**